reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/DisposableWrapper.kt

```diff
@@ -24,10 +25,21 @@ class DisposableWrapper : Disposable {
 
     private fun setHolder(holder: Holder?) {
         ref
-            .getAndSet(holder)
-            ?.disposable
+            .getAndUpdate { oldHolder ->
+                if (oldHolder == null) {
+                    holder?.dispose()
+                }
+
+                holder
+            }
             ?.dispose()
     }
 
     private class Holder(val disposable: Disposable?)
+
+    private companion object {
+        private fun Holder.dispose() {
```

Comment:
Why extension function? You have access to `Holder` class, create dispose function there.